tools/src/main/java/org/apache/kafka/trogdor/workload/ConnectionStressWorker.java

```diff
@@ -250,6 +254,9 @@ public void stop(Platform platform) throws Exception {
         doneFuture.complete("");
         workerExecutor.shutdownNow();
         workerExecutor.awaitTermination(1, TimeUnit.DAYS);
+        synchronized (ConnectionStressWorker.this) {
```

Comment:
Even though calling `updateStatus()` here is thread-safe, I think that using the lock here acts as a memory barrier and allows us to access the latest values of the variables. I very well might be wrong though and this may be superfluous

Reply:
It's not thread-safe without the `synchronized`, since we're accessing variables that need synchronization.  But let's move the `synchronized` keyword into `ConnectionStressWorker#updateStatus`, as described above, since that's a bit cleaner